### ISSUE TITLE
`cdktf plan` throws axios maxBodyLength error

### ISSUE THREAD

**Issue by jaismith** [hashicorp/terraform-cdk#1470]
### Community Note

- Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
- Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
- If you are interested in working on this issue or have submitted a pull request, please leave a comment

### cdktf & Language Versions

- "@cdktf/provider-google": "^0.5.128",
- "cdktf": "^0.8.5",
- "cdktf-cli": "^0.8.5",
- "typescript": "^4.5.4"
- "node": "16.13.1"

### Debug Output

[gist](https://gist.github.com/jaismith/f36b6ceec80ecb7ed696aed687a45845)

### Expected Behavior

- should generate plan

### Actual Behavior

- throws error, looks like an axios maxBodyLength error

### Steps to Reproduce

Not sure what in particular is causing this, this used to periodically happen when deploying changes and I cleared node_modules/.gen and seemed to be able to fix it, though not 100% sure that was actually what was fixing it. Now I can't seem to get it to work, I've upgraded all my dependencies to their latest versions, including yarn.

### Important Factoids

- using terraform cloud, the network call causing this error may be related to that

### References

- [this](https://github.com/axios/axios/issues/1362) seems to be talking about the same error, though not terraform related. wondering if `maxBodyLength` needs to be increased on an internal cdktf axios call

**Comment by jaismith** [comment]
In an effort to resolve this over the weekend, I uninstalled and reinstalled the terraform cli via brew, which prompted me to update from MacOS 12.0.1 to 12.1 (some Xcode Command Line Tools update). After updating, and reinstalling the terraform CLI, I'm no longer seeing this issue. Not sure if it was related to the update, or a corrupted cache somewhere?

TLDR: Updated MacOS 12.0.1 -> 12.1 and reinstalled terraform via brew, issue no longer presents.

**Comment by jaybocc2** [comment]
also seeing this.  Using cdktf `0.10.1` and using terraform cloud
```
136,182,42,130,16,243,24,95,170,58,128,157,92,167,6,148,163,66,128,123,152,59,13,211,123,143,238,205,143,209,146,171,243,103,23,222,145,70,26,105,164,79,166,127,125,101,61,16,0,40,221,18]},"headers":{"Accept":"application/json, text/plain, */*","Content-Type":"application/x-www-form-urlencoded","User-Agent":"axios/0.21.4","Content-Length":60332997},"transformRequest":[null],"transformResponse":[null],"timeout":0,"xsrfCookieName":"XSRF-TOKEN","xsrfHeaderName":"X-XSRF-TOKEN","maxContentLength":-1,"maxBodyLength":-1,"transitional":{"silentJSONParsing":true,"forcedJSONParsing":true,"clarifyTimeoutError":false}},"code":"ERR_FR_MAX_BODY_LENGTH_EXCEEDED"}
Request body larger than maxBodyLength limit
```
the numbers are just a byte array i assume and it is over 10,000 lines of the byte array, so this is all i can really provide.

if i manually `terraform plan/apply` it works fine.  Seems to happen after the the cdk output gets zipped up and uploaded.

**Comment by ansgarm** [comment]
Hi @jaybocc2 👋
Are there any particularly large files in `cdktf.out/stacks/jtest-vpc`? Could you send me some file tree with filesizes?
What kinds of resources are you using in your stack? Are you using large Assets?

**Comment by jaybocc2** [comment]
Hi @ansgarm  good callout, i had i guess at some point ran terraform manually creating a .terraform directory that had AWS provider there which is 300mb.

cleaning this out and everything is working fine.

Apologies for the fire drill.

**Comment by ansgarm** [comment]
Nevermind! This is something we should improve (e.g. explicitly ignoring the `.terraform` directory in there) 👍

**Comment by ansgarm** [comment]
@DanielMSchmidt you're right, but that method had a bug 😁 see #1694.

Here's why:
https://github.com/hashicorp/terraform-cdk/blob/dc752a93a3f7c264c00771b3668d89a882ebd70b/packages/cdktf/lib/manifest.ts#L49-L54
https://github.com/hashicorp/terraform-cdk/blob/dc752a93a3f7c264c00771b3668d89a882ebd70b/packages/cdktf/lib/manifest.ts#L29

**Comment by DanielMSchmidt** [comment]
Nice catch 👍

**Comment by DanielMSchmidt** [comment]
Closing this for now since we removed the function clearing out the local terraform providers
